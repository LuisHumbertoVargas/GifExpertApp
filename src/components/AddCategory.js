import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        // console.log('handle Submit', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories((categories) => [inputValue, ...categories]);
            setInputValue('');
        }
    };

    const handleMouseDown = () => {
        setInputValue('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>Insert category</p>
            <input
                id={'category'}
                type='text'
                value={inputValue}
                onChange={handleInputChange}
                onMouseDown={handleMouseDown}
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
