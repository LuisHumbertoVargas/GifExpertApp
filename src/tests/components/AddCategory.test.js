import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Prueba unitaria en el componente <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Debe de retornar un snapshot del componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar el estado del input al ingresar texto', () => {
        const input = wrapper.find('#category');
        const value = 'Insert category';
        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value); // toBe( type: string )
    });

    test('NO debe postear la informaciÃ³n cuando se ejecute el submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() {} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        // 1.- Simular el inputChange
        const input = wrapper.find('#category');
        const value = 'Space X';
        input.simulate('change', { target: { value } });

        // 2.- Simular el inputSubmit
        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault() {} });

        // 3.- setCategories debe ser llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        // 4.- El valor del input debe de estar vacio ''
        expect(input.prop('value')).toBe('');
    });
});
