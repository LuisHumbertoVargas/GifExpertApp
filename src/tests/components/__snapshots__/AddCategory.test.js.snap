// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba unitaria en el componente <AddCategory /> Debe de retornar un snapshot del componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Insert category
  </p>
  <input
    id="category"
    onChange={[Function]}
    onMouseDown={[Function]}
    type="text"
    value=""
  />
</form>
`;
